src/components/ScreenAuthors.vue

```diff
@@ -107,40 +107,27 @@ export default defineComponent({
         const { errors } = useValidation()
         const editingId = ref(0)
         const addAuthor = async () => {
-            let newAuthors:AuthorType[]
             const newAuthor: AuthorType = {}
-            if (authors.value === undefined) {
-                newAuthors = [newAuthor]
-            } else {
-                newAuthors = [...authors.value, newAuthor]
-            }
+            const newAuthors = [...authors.value, newAuthor]
             setAuthors(newAuthors)
             editingId.value = newAuthors.length - 1
             // await the DOM update that resulted from updating the authors list
             await nextTick()
             scrollToBottom()
         }
         const removeAuthor = () => {
-            if (authors.value !== undefined) {
-                const newAuthors = [...authors.value]
-                newAuthors.splice(editingId.value, 1)
-                setAuthors(newAuthors)
-                editingId.value = -1
-                if (Array.isArray(newAuthors) && newAuthors.length === 0) {
-                    setAuthors([])
-                }
-            }
+            const newAuthors = [...authors.value]
+            newAuthors.splice(editingId.value, 1)
+            setAuthors(newAuthors)
+            editingId.value = -1
         }
         const setAuthorField = (field: keyof AuthorType, value: string) => {
-            if (authors.value !== undefined) {
-                const author = { ...authors.value[editingId.value] }
-                author[field] = value === '' ? undefined : value
-                authors.value[editingId.value] = author
-                setAuthors(authors.value)
-            }
+            const author = { ...authors.value[editingId.value] }
```

Comment:
Should this have been called `newAuthor` (since it is a reference to an object that did not previously exist)?